libraries/rush-lib/src/logic/ProjectChangeAnalyzer.ts

```diff
@@ -402,58 +315,178 @@ export class ProjectChangeAnalyzer {
+        if (!(await FileSystem.existsAsync(filePath))) {
+          throw new Error(
+            `A project dependency file (${filePath}) is missing. You may need to run ` +
+              '"rush install" or "rush update".'
+          );
+        }
+      });
+    } else {
+      // Determine the current variant from the link JSON.
+      const variant: string | undefined = this._rushConfiguration.currentInstalledVariant;
+
+      // Add the shrinkwrap file to every project's dependencies
+      const shrinkwrapFile: string = Path.convertToSlashes(
+        path.relative(rootDirectory, this._rushConfiguration.getCommittedShrinkwrapFilename(variant))
       );
 
-      return undefined;
+      globalAdditionalFiles.push(shrinkwrapFile);
     }
+
+    return {
+      additionalGlobs,
+      additionalRelativePathsToHash,
+      gitPath,
+      globalAdditionalFiles,
+      projectMap,
+      rootDirectory
+    };
   }
 }
+
+interface IAdditionalGlob {
+  project: RushConfigurationProject;
+  operationName: string;
+  additionalFilesForOperation: Set<string>;
+  pattern: string;
+}
+
+async function getAdditionalFilesFromRushProjectConfigurationAsync(
+  additionalGlobs: IAdditionalGlob[],
+  lookupByPath: LookupByPath<RushConfigurationProject>,
+  terminal: ITerminal
+): Promise<Set<string>> {
+  const additionalFilesFromRushProjectConfiguration: Set<string> = new Set();
+
+  if (!additionalGlobs.length) {
+    return additionalFilesFromRushProjectConfiguration;
+  }
+
+  const { default: glob } = await import('fast-glob');
+  await Async.forEachAsync(additionalGlobs, async (item: IAdditionalGlob) => {
+    const { project, operationName, additionalFilesForOperation, pattern } = item;
+    const matches: string[] = await glob(pattern, {
+      cwd: project.projectFolder,
+      onlyFiles: true,
+      // We want to keep path's type unchanged,
+      // i.e. if the pattern was a  relative path, then matched paths should also be relative paths
+      //      if the pattern was an absolute path, then matched paths should also be absolute paths
+      //
+      // We are doing this because these paths are going to be used to calculate operation state hashes and some users
+      // might choose to depend on global files (e.g. `/etc/os-release`) and some might choose to depend on local non-project files
+      // (e.g. `../path/to/workspace/file`)
+      //
+      // In both cases we want that path to the resource to be the same on all machines,
+      // regardless of what is the current working directory.
+      //
+      // That being said, we want to keep `absolute` options here as false:
+      absolute: false
+    });
+
+    for (const match of matches) {
+      if (path.isAbsolute(match)) {
+        additionalFilesFromRushProjectConfiguration.add(match);
+        additionalFilesForOperation.add(match);
+      } else {
+        const filePathForGit: string = path.posix.join(project.projectRelativeFolder, match);
```

Comment:
This will be from the root of the rush.json folder. Should verify if that is the correct path root for the git hash for files method.

Reply:
Any reason to not just concatenate the path parts together with a slash?